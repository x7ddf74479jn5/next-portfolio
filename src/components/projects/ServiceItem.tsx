import clsx from "clsx";
import React from "react";
import type { Service } from "src/contents/services";
import useIntersection from "src/hooks/useIntersection";
import styles from "src/styles/components/projects/ServiceItem.module.scss";

type Props = {
  animation?: "slideInLeft" | "slideInRight" | "";
  service: Service;
};

const ServiceItem: React.VFC<Props> = ({ animation, service }) => {
  const ref = React.useRef(null);
  const isIntersect = useIntersection(ref, { root: null, rootMargin: "0px", threshold: 0 });
  const className = animation
    ? clsx(styles.gridListItemFloated, { [styles[animation]]: isIntersect })
    : styles.gridListItemFloated;

  return (
    <article className={className} ref={ref}>
      <h3>{service.title}</h3>
      <small>{service.caption}</small>
      <img className={styles.mediaThumb} src={service.img.src} alt={service.img.alt} width={512} height={288} />
      <p>{service.description}</p>
    </article>
  );
};

export default ServiceItem;
