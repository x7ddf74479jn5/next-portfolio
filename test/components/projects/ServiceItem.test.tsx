import ServiceItem from "src/components/projects/ServiceItem";
import type { RenderResult } from "test/test-utils";
import { reTestCase } from "test/test-utils";
import { render } from "test/test-utils";

import type { Service } from "../../../src/contents/services";

describe("ServiceItem", () => {
  let renderResult: RenderResult;

  const service: Service = {
    title: "title",
    caption: "caption",
    img: {
      src: "src.png",
      alt: "alt",
    },
    description: "description",
  };

  it("matches snapshot", () => {
    renderResult = render(<ServiceItem animation="slideInLeft" service={service} />);
    expect(renderResult.asFragment()).toMatchSnapshot();
  });

  it("rendered correctly", () => {
    renderResult = render(<ServiceItem animation="slideInLeft" service={service} />);

    expect(renderResult.getByRole("heading")).toHaveTextContent("title");

    expect(renderResult.getByRole("img")).toHaveAttribute("src", reTestCase.anyImage);
    expect(renderResult.getByRole("img")).toHaveAttribute("src", reTestCase.anyWord);
  });
});
