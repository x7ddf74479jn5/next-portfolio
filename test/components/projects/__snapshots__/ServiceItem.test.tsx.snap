// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceItem matches snapshot 1`] = `
<DocumentFragment>
  <article
    class="gridListItemFloated"
  >
    <h3>
      title
    </h3>
    <small>
      caption
    </small>
    <img
      alt="alt"
      class="mediaThumb"
      height="288"
      src="src.png"
      width="512"
    />
    <p>
      description
    </p>
  </article>
</DocumentFragment>
`;
